popup.js

```diff
@@ -0,0 +1,156 @@
+
+let currentVid;
+let current = [];
+
+// adding a new bookmark row to the popup
+const addNewBm = (bms, bookmark) => {
+
+	// bookmark title element
+	const bmTitle = document.createElement('div');
+	bmTitle.textContent = bookmark.desc;
+	bmTitle.className = 'bookmark-title';
+
+	const editBmTitle = document.createElement('img');
+	editBmTitle.src = 'assets/edit.png';
+	editBmTitle.title = 'Edit bookmark Title';
+	editBmTitle.addEventListener('click', onEdit);
+
+	const playBm = document.createElement('img');
+	playBm.src = 'assets/play.png';
+	playBm.title = 'Start play from bookmark';
+	playBm.addEventListener('click', onPlay);
+
+	const deleteBm = document.createElement('img');
+	deleteBm.src = 'assets/delete.png';
+	deleteBm.title = 'Delete bookmark';
+	deleteBm.addEventListener('click', onDelete);
+	
+	// bookmark controls - edit bookmark title, play from bookmark
+	const bmControls = document.createElement('div');
+	bmControls.className = 'bookmark-controls';
+
+	bmControls.appendChild(editBmTitle);
+	bmControls.appendChild(playBm);
+	bmControls.appendChild(deleteBm);
+
+	const bm = document.createElement('div');
+	bm.id = 'bookmark-'+bookmark.time;
+	bm.className = 'bookmark';
+	bm.setAttribute('timestamp', bookmark.time);
+	
+	bm.appendChild(bmTitle);
+	bm.appendChild(bmControls);
+	bms.appendChild(bm);
+}
+
+const viewBms = (bmsList) => {
+	const bms = document.getElementById("bookmarks");
+	bms.innerHTML = ''; // removes anything in the previous list
+
+	if (bmsList && bmsList.length > 0) {
+		bmsList.forEach((bookmark) => {
```

Comment:
would challenge you to not use forEach functions and use a for in or a for of depending on if this is an object or an array

Reply:
the forEach is the slowest js iterator

Reply:
sure, added for here